armi/utils/plotting.py

```diff
@@ -1047,3 +1063,304 @@ def makeHistogram(x, y):
         yHistogram[upper] = y[i]
     xHistogram[0] = x[0] / 2.0
     return xHistogram, yHistogram
+
+
+def _makeBlockPinPatches(block, cold):
+    """Return lists of block component patches and corresponding data and names (which relates to material
+    of the component for later plot-coloring/legend) for a single block.
+
+
+    Takes in a block that must have a spatialGrid attached as well as a variable
+    which signifies whether the dimensions of the components are at hot or cold temps.
+    When cold is set to true, you would get the BOL cold temp dimensions.
+
+    Parameters
+    ----------
+    block : Block
+
+    cold : boolean
+        true for cold temps, hot = false
+
+    Return
+    ------
+    patches : list
+        list of patches for block components
+
+    data : list
+        list of the materials these components are made of
+
+    name : list
+        list of the names of these components
+    """
+    from armi.reactor.components import DerivedShape
+
+    patches = []
+    data = []
+    name = []
+    if isinstance(block.spatialGrid, grids.HexGrid):
+        largestPitch, comp = block.getPitch(returnComp=True)
+
+    elif isinstance(block.spatialGrid, grids.ThetaRZGrid):
+        raise TypeError(
+            "This plot function is not currently supported for ThetaRZGrid grids."
+        )
+    else:
+        largestPitch, comp = block.getPitch(returnComp=True)
+        if block.getPitch()[0] != block.getPitch()[1]:
+            raise ValueError("Only works for blocks with equal length and width.")
+
+    sortedComp = [c for c in block]
+
+    derivedComponent = block.getComponentsOfShape(DerivedShape)
+    if len(derivedComponent) == 1:
+        component = derivedComponent[0]
+        cName = component.name
+        material = component.material.name
+        if isinstance(comp, Hexagon):
+            derivedPatch = matplotlib.patches.RegularPolygon(
+                (0, 0), 6, largestPitch / math.sqrt(3)
+            )
+        elif isinstance(comp, Square):
+            derivedPatch = matplotlib.patches.Rectangle(
+                (0, 0), largestPitch[0], largestPitch[0]
+            )
+        else:
+            raise TypeError(
+                "Shape of the pitch-defining element is not a Square or Hex it is {}, cannot plot for this type of block".format(
+                    comp.shape
+                )
+            )
+        patches.append(derivedPatch)
+        data.append(material)
+        name.append(cName)
+    sortedComp.sort(reverse=True)
+    for components in sortedComp:
+        if components.name is not "pitch" and not isinstance(components, DerivedShape):
```

Comment:
Maybe just `.remove()` the DerivedShape component from `sortedComp[s]` up above so that you don't need to keep checking.

Also, why is the "pitch" case special?

Reply:
"pitch" is something that is considered a "dummy component for assembly sizing" in the Cartesian grid file I was using. It had a material called Void, but I just noticed Inner and Outer Width are equal for it, so I think I would be able to still draw it and not actually have it show up without checking. But then if I do not have some check, Void,  it's material that may not be used by any other component would still show up in the legend even if it wasn't actually present in the diagram. But would a "dummy component for assembly sizing" always have the same name, or not?